### PR TITLE
Seperate shim.conf and jars dependency and fix alarm scheduling

### PR DESCRIPTION
**Pull request by homezcx**
1. Seperate shim.conf and jars dependency.
2. Reduce the alarm spamming messages.
3. Change the timeout value to 5 min to match up YARN setup.
4. Remove soft link in LinuxCommand Builder, as it will throw error if "reef" directory exists, even with "sfn" option set.
5. Fix an exception when Shim is trying to create a directory when extracting files.

With this change, the performance of CloseEvaluatorTest (16 evaluators) improves from 5 min + to 78 seconds.

### CODE REVIEW

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimManager.java [hunk 1]
```diff
@@ -303,7 +300,7 @@ private File writeFileResourcesJarFile(final Set<FileResource> fileResourceSet)
 
   private URI uploadFile(final File jarFile) throws IOException {
     final String folderName = this.azureBatchFileNames.getStorageJobFolder(this.getAzureBatchJobId());
-    LOG.log(Level.FINE, "Uploading {0} to {0}.", new Object[] {jarFile.getAbsolutePath(), folderName});
+    LOG.log(Level.FINE, "Uploading {0} to {0}.", new Object[]{jarFile.getAbsolutePath(), folderName});
```
Comment (ryumbra) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimManager.java]: Uploading {0} to {1}.

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchResourceManager.java [hunk 2]
```diff
@@ -72,13 +73,19 @@
   }
 
   public void onResourceRequested(final ResourceRequestEvent resourceRequestEvent) {
-    LOG.log(Level.FINEST, "Got ResourceRequestEvent in AzureBatchResourceManager,");
+    LOG.log(Level.INFO, "Got ResourceRequestEvent in AzureBatchResourceManager,");
+    URI jarFileUri = this.evaluatorShimManager.generateShimJarFile();
     for (int r = 0; r < resourceRequestEvent.getResourceCount(); r++) {
       final String containerId = generateContainerId();
+      LOG.log(Level.INFO, "containerId in AzureBatchResourceManager {0}", containerId);
```
Comment (ryumbra) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchResourceManager.java]: This can probably be downgraded to FINE or FINER.

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchResourceManager.java [hunk 3]
```diff
@@ -72,13 +73,19 @@
   }
 
   public void onResourceRequested(final ResourceRequestEvent resourceRequestEvent) {
-    LOG.log(Level.FINEST, "Got ResourceRequestEvent in AzureBatchResourceManager,");
+    LOG.log(Level.INFO, "Got ResourceRequestEvent in AzureBatchResourceManager,");
```
Comment (ryumbra) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchResourceManager.java]: Any reason to up this to INFO?

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/AzureBatchFileNames.java [hunk 4]
```diff
@@ -27,7 +27,7 @@
  * Access to the various places things go according to the REEF Azure Batch runtime.
  */
 @Private
-public final class AzureBatchFileNames {
+public final class AzureBatchFileNames extends REEFFileNames {
```
Comment (ryumbra) [inline lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/AzureBatchFileNames.java]: Why is this change necessary?

**Review by ryumbra — APPROVED** [review]

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchResourceManager.java (hunk 3 above)
Comment (homezcx) [reply lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchResourceManager.java]: No... will fix these logs.

File: lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/AzureBatchFileNames.java (hunk 4 above)
Comment (homezcx) [reply lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/AzureBatchFileNames.java]: So I could call this.azureBatchFileNames.getLocalFolderPath() in ShimManager without Injecting REEFFileNames. But I just noticed it's already there. I will revert it .
